src/components/store/StoreProvider.tsx

```diff
@@ -1,16 +1,18 @@
-import { useEffect } from "react";
-import { StoreContext, store } from "./store.tsx";
+import { useEffect, useRef } from "react";
+import { StoreContext, createMyStore } from "./store.tsx";
 
 export function GlobalStateProvider({
   children,
 }: {
   children: React.ReactNode;
 }) {
+  const store = useRef(createMyStore());
```

Comment:
Tady jsem to jen lehce predelal, at ten stav vytvarime on demand, a ne automaticky v module, to se nam muze hodit treba v testech, muzeme pak zkusit napsat nejakej sofistikovanejsi test, co nam otestuje i nejakou logiku v tech actions/selectorech.